### PR TITLE
added option to rocdl to build hc lib & cleanup

### PR DESCRIPTION
**Pull request by dfukalov**
1. added option to rocdl to build hc lib
2. minor cleanup of unused include

### CODE REVIEW

**Comment by dfukalov** [comment]
the option is "on" by default in device library currently, but going to be changed to default value "off" in order to re-enable the library build with LC (not HCC)

File: CMakeLists.txt
Comment (whchung) [inline CMakeLists.txt:417]: @dfukalov there is no such flag `BUILD_HC_LIB` inside `ROCm-Device-Libs`, could you explain the rationale behind adding this line?

**Review by whchung — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (whchung) [reply CMakeLists.txt:417]: @dfukalov after checking your earlier comment I think I know the rationale now. I'll hold off this PR and merge it when `ROCm-Device-Libs` does carry this switch.

**Review by whchung — APPROVED** [review]

File: CMakeLists.txt
Comment (dfukalov) [reply CMakeLists.txt:417]: Yes, the switch was added in http://git.amd.com:8080/#/c/82189/ and is not arrived yet.
The will not brake build, however warning of unused variable will be produced.

**Comment by dfukalov** [comment]
will merge this pull request once library change arrives to hcc tree

**Pull request merged**
